frontend/src/components/Profile/Profile.tsx

```diff
@@ -32,122 +45,162 @@ const InnerProfile: React.FC = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    if (user && user.user_metadata) {
+      setFormData((prev) => ({
+        ...prev,
+        ...user.user_metadata as UserMetadata,
+      }));
+    }
+    if (user && user.email) {
+      setFormData((prev) => ({ ...prev, email: user.email as string }));
+    }
+  }, [user]);
+
+  const handleSave = async () => {
+    try {
```

Comment:
What's the try/catch for? That should be done internally by the supabase function right?

Reply:
I changed to "const { data, error } = await client.auth.updateUser({ data: formData });"